controllers/numaresourcesoperator_controller.go

```diff
@@ -222,29 +222,34 @@ func (r *NUMAResourcesOperatorReconciler) reconcileResourceMachineConfig(ctx con
 	// It can take a while.
 	mcpStatuses, allMCPsUpdated := syncMachineConfigPoolsStatuses(instance.Name, trees, r.ForwardMCPConds)
 	instance.Status.MachineConfigPools = mcpStatuses
+	instance.Status.NodeGroups = syncNodeGroupsStatusPerMCPWithOperatorStatus(instance, false)
```

Comment:
sorry for my lack of understanding, but why do we need to call this function twice (here and in line 232) and one time with `false` and then with `true`? 
this deserve at least a comment IMO

Reply:
good catch. This deserves a comment and another honest attempt at simplifying the code.

Reply:
good point. I was following the rules by which the MCP status would get updated, which is only if the MC is updated then update the corresponding NodeGroupConfig in the status, hence the forwarded boolean value.
I agree the code in this area needs can be improved and also adding a way to test the config status update in the controller tests. I'll open an issue to track this work.

Reply:
Glancing at the open issues, I see this:https://github.com/openshift-kni/numaresources-operator/issues/1031 which sounds to me it includes the subject raised here?